### PR TITLE
Allow to hide in which headers the struct is defined

### LINKED ISSUE #57: Unexpected include in the documentation of struct within a namespace
Running the testsuite on the latest version of breathe (b4d53b9...) produces the following output for `foo::ns::FooStruct`:

```
struct foo::ns::FooStruct
#include <namespacefile.h>
     Documentation here.
```

Note the line `"#include <namespacefile.h>"`. It is not in the example on the website: http://michaeljones.github.com/breathe/struct.html#example-with-namespace and the documentation of a struct without a namespace (in global scope) doesn't have such line either.

I am using doxygen 1.8.1.2.

### PR DESCRIPTION
**Pull request by rhssk**
When something like "doxygennamespace" is used, structures gain
additional line about in which header they are defined.

If the API simply serves as a generic description and no public headers
are provided, this info is redundant. This can be worked around
manually by using "doxygenstruct" directive instead, but this can get
messy quickly with large namespaces. Also, namespace description is
lost.

Allow to hide these lines by setting 'breathe_show_struct_include' to
'False'. The default behavior stays unchanged.

Fixes #57.

### CODE REVIEW

File: documentation/source/directives.rst [hunk 1]
```diff
@@ -562,6 +562,22 @@ Config Values
    For example an enum value ``TWO = 2`` would be shown with the value 2 if this is set to ``True``,
    and without if it is set to ``False``.
 
+.. confval:: breathe_show_struct_include
```
Comment (vermeeren) [inline documentation/source/directives.rst:565]: @rhssk Thanks for the patch, overall looks good to me.

I'm not sure about the naming though, the way things are implemented it seems like `visit_inc` is a generic function not specific to struct, although perhaps current implementations only renders this on struct?

Thoughts on renaming the option to just `breathe_show_include`?

Comment (jakobandersen) [reply documentation/source/directives.rst:565]: +1 for renaming to  ``breathe_show_include``.

File: breathe/directives.py [hunk 2]
```diff
@@ -587,6 +587,7 @@ def setup(app: Sphinx) -> None:
     app.add_config_value("breathe_default_members", (), True)
     app.add_config_value("breathe_show_define_initializer", False, 'env')
     app.add_config_value("breathe_show_enumvalue_initializer", False, 'env')
+    app.add_config_value("breathe_show_struct_include", True, 'env')
```
Comment (jakobandersen) [inline breathe/directives.py:590]:
```suggestion
    app.add_config_value("breathe_show_include", True, 'env')
```

File: breathe/renderer/sphinxrenderer.py [hunk 3]
```diff
@@ -1651,6 +1651,9 @@ def visit_verbatim(self, node) -> List[Node]:
         return [rst_node]
 
     def visit_inc(self, node) -> List[Node]:
+        if not self.app.config.breathe_show_struct_include:
```
Comment (jakobandersen) [inline breathe/renderer/sphinxrenderer.py:1654]:
```suggestion
        if not self.app.config.breathe_show_include:
```

File: documentation/source/directives.rst (hunk 1 above)
Comment (jakobandersen) [inline documentation/source/directives.rst:565]:
```suggestion
.. confval:: breathe_show_include
```

File: documentation/source/directives.rst [hunk 4]
```diff
@@ -562,6 +562,22 @@ Config Values
...
+
+   A boolean flag which can be set to ``False`` to hide the header in which the
```
Comment (jakobandersen) [inline documentation/source/directives.rst:567]:
```suggestion
   A boolean flag which can be set to ``False`` to hide the header in which each
```

File: documentation/source/directives.rst [hunk 5]
```diff
@@ -562,6 +562,22 @@ Config Values
...
+   structure is defined.
```
Comment (jakobandersen) [inline documentation/source/directives.rst:568]:
```suggestion
   entity (struct, function, macro, etc.) is defined.
```

File: documentation/source/directives.rst [hunk 6]
```diff
@@ -562,6 +562,22 @@ Config Values
...
+
+   When set to ``True`` (default)::
```
Comment (jakobandersen) [inline documentation/source/directives.rst:570]:
```suggestion
   For example, when set to ``True`` (the default) a ``struct Foo`` is rendered similarly to::
```

File: documentation/source/directives.rst [hunk 7]
```diff
@@ -562,6 +562,22 @@ Config Values
...
+
+     struct Foo
+       #include <foo.hpp>
+       Description of Foo.
+
+   When set to ``False``::
```
Comment (jakobandersen) [inline documentation/source/directives.rst:576]:
```suggestion
   but when set to ``False`` it is instead rendered as::
```

**Review by jakobandersen — CHANGES_REQUESTED** [review]
Changes for renaming to ``breathe_show_include`` made as suggestions in Github.

**Comment by rhssk** [comment]
Updated with the requested changes.

**Comment by rhssk** [comment]
And rebased on `master`

**Comment by 2bndy5** [comment]
@rhssk Can you rebase this branch on master?

@jakobandersen  @vermeeren It looks like all requested changes were applied. Any further objections to merging this (after rebase is done)?

**Review by jakobandersen — APPROVED** [review]

**Comment by jakobandersen** [comment]
Thanks! Merged via rebase (87c6ca1dcd0558ec884e01eb8ea71a7913c73c06 + parent).

**Pull request closed without merging**